### ISSUE TITLE
Allow to show today's current events in the upcoming events

### ISSUE THREAD

**Issue by k-webdesign** [contao/contao#2455]
When you set the event list format to "all events" you can't see the events that are happening today. I found a workaround to hide each event at 23:59 at the end of the day but it would be much more comfortable if we had an option to include today's event(s) into the "all events" list.

**Comment by asaage** [comment]
Todays events should be included in all events.
If you just need today's events go:
![image](https://user-images.githubusercontent.com/4138336/96998066-2993ce80-1533-11eb-962a-5837764a6c5d.png)
What exactly are you trying to accomplish?

**Comment by k-webdesign** [comment]
I don't want to show just today's events, I want to show all upcoming events including today's. But "today" becomes invisible as soon as it's 00:01 (or even 00:00, I don't know).

**Comment by asaage** [comment]
I can't reproduce that in the demo 🤷‍♂️


![image](https://user-images.githubusercontent.com/4138336/97295605-d03be000-184f-11eb-96a6-d2d601faaa06.png)
--
![image](https://user-images.githubusercontent.com/4138336/97295366-7804de00-184f-11eb-8265-465c7af6a279.png)
![image](https://user-images.githubusercontent.com/4138336/97295423-8b17ae00-184f-11eb-8e7b-8ccf9e777100.png)
--
![image](https://user-images.githubusercontent.com/4138336/97295530-b26e7b00-184f-11eb-9b49-19e7dff99c8d.png)
--

>  I found a workaround to hide each event at 23:59 at the end of the day

maybe the problem is in your "workaround"?

**Comment by fritzmg** [comment]
I can reproduce the problem locally in Contao 4.9.8.

**Comment by fritzmg** [comment]
I can also reproduce the problem in the Contao Demo.

**Comment by fritzmg** [comment]
The difference between @asaage 's Test and mine is probably the `tl_module.cal_noSpan` setting (**Shortend view**). I disabled it, while @asaage probably kept it enabled. If you disable `tl_module.cal_noSpan`, then the event list showing all upcoming events will not display today's event. If you enable it, then it will display today's event.

**Comment by asaage** [comment]
Yes **(Shortened view)**-disabled seems to cause that unexpected behaviour.

**Comment by asaage** [comment]
I think this is actually a duplicate of https://github.com/contao/contao/issues/2448

**Comment by fritzmg** [comment]
Hm, this is a tough one to fix. Technically I would say, that the event showing up when **Shortened view** is enabled is a bug, since it is not a future event any more.

However I do find it desirable to always list events of the current day. May be this would be another checkbox? 😬 **Always show events of current day**. But then this would also list events of the current day, which have an end time in the past …

**Comment by k-webdesign** [comment]
I also think it's a bug... 

In which case have events of the current day an end time in the past? They are current when they're **today** and end somewhen today.

**Comment by aschempp** [comment]
It shouldn't be a checkbox, it's simply another option to "Show today and future events" imho.

**Comment by Ernestopheles** [comment]
In my case, the event will be shown on the same day as well, but only before start time. As soon as start time is reached, the event will not be shown any more.
using this module: 
![Screenshot_20210423_165336](https://user-images.githubusercontent.com/7561403/115889747-80338400-a454-11eb-838d-0ee3eb501c2b.png)

My wish is, this event should be shown until end time

**Comment by sietsevandreven** [comment]
I came across this issue again. Unfortunately it hasn't been fixed yet. How I expect it to work is that an event becomes invisible when the end time is passed. If no time is specified, the current day's event should be visible until the day is over. If the end time is passed the event should be visible in the past events setting.

It's a really annoying problem and I hope it can be fixed soon.
